walterthesmart/LuxuryGoodss
# User Interface build

# Luxury NFT Store User Interface

## Overview
This pull request introduces a new User Interface (UI) for our Luxury NFT store. The Luxury Store is a platform where users can purchase luxury items and receive an NFT transferred to their wallet upon completion of the purchase. The smart contract for this store is written in Clarity and deployed on the Stacks blockchain.

## Changes Introduced
- Created a new React component (`App.js`) that serves as the main entry point for the UI.
- Implemented wallet connection functionality using `@stacks/connect`.
- Added a basic navigation bar with the store title and a wallet connection button.
- Implemented state management for user wallet address using React's `useState` hook and browser's `sessionStorage`.

## Technical Details

### Dependencies
- `@stacks/connect`: Used for Stacks blockchain integration and wallet connection.
- React: The UI is built using React, leveraging hooks for state management.

### Component Structure
The `App` component is the main component of the application. It includes:
- Configuration for Stacks connection (`AppConfig` and `UserSession`).
- State management for user wallet address.
- A `handleSubmit` function to manage wallet connection.
- A simple UI with a navigation bar and wallet connection button.

### Wallet Connection
- The `showConnect` function from `@stacks/connect` is used to initiate the wallet connection process.
- Upon successful connection, the user's testnet STX address is stored in both component state and `sessionStorage`.

### UI Elements
- The UI currently consists of a navigation bar with the store title "Luxury Goods UI" and a dynamic button for wallet connection.
- Tailwind CSS classes are used for styling, providing a responsive and modern look.

## Next Steps
1. Implement the main store interface for browsing and purchasing luxury items.
2. Integrate with the Clarity smart contract to handle NFT transfers upon purchase.
3. Add error handling and loading states for wallet connection.
4. Implement a more comprehensive state management solution if the application grows in complexity.
5. Add unit and integration tests for the React components and blockchain interactions.

## Testing Instructions
1. Clone the repository and install dependencies.
2. Run the application locally.
3. Click the "Connect Wallet" button and ensure you can connect your Stacks wallet.
4. Verify that the button text changes to "Wallet Connected" after successful connection.
5. Check that the wallet address persists after page refresh due to `sessionStorage` usage.


Please review and let me know if any changes or additions are needed!
